internal/kafka/internal/services/kafkatlscertmgmt/secure_storage.go

```diff
@@ -12,19 +12,23 @@ import (
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
 	"github.com/aws/aws-secretsmanager-caching-go/secretcache"
 	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
 	"github.com/caddyserver/certmagic"
 )
 
+//go:generate moq -out secretmanager_client_moq.go . SecretManagerClient
+type SecretManagerClient = secretsmanageriface.SecretsManagerAPI
```

Comment:
This is a nice approach, and it makes me think that it should be possible to to the same for other external types that we are generating workarounds for. Specifically, for Gocloak and Route53 where we call the moq command referencing the vendor directories.

We could define type alias for them and generate them instead from our packages, removing the generation workarounds.
What do you think? If so, I can take a look a it.

Reply:
+1 on that. I think it'll also remove the need for vendoring. Thank you for having a look.

Reply:
@pb82 this is done https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/Makefile#L417-L418

Reply:
I will work on generating those mocks locally